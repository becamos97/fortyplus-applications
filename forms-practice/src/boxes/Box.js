import React from "react";

export default function Box({ width, height, backgroundColor, removeBox }) {
  const style = {
    width: `${width}px`,
    height: `${height}px`,
    backgroundColor,
    display: "inline-block",
    margin: 8,
    borderRadius: 6,
  };
  return (
    <div style={{ display: "inline-flex", alignItems: "center" }}>
      <div data-testid="box" style={style} />
      <button aria-label="remove-box" onClick={removeBox}>X</button>
    </div>
  );
}