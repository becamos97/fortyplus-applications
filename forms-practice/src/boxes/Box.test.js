import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import Box from "./Box";
import "@testing-library/jest-dom";

it("renders without crashing", () => {
  render(<Box width={50} height={60} backgroundColor="red" removeBox={() => {}} />);
});

it("matches snapshot", () => {
  const { asFragment } = render(<Box width={50} height={60} backgroundColor="red" removeBox={() => {}} />);
  expect(asFragment()).toMatchSnapshot();
});

it("calls remove on click", () => {
  const mock = jest.fn();
  render(<Box width={50} height={60} backgroundColor="red" removeBox={mock} />);
  fireEvent.click(screen.getByLabelText(/remove-box/i));
  expect(mock).toHaveBeenCalled();
});