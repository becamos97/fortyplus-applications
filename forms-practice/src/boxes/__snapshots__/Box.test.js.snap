// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    style="display: inline-flex; align-items: center;"
  >
    <div
      data-testid="box"
      style="width: 50px; height: 60px; background-color: red; display: inline-block; margin: 8px; border-radius: 6px;"
    />
    <button
      aria-label="remove-box"
    >
      X
    </button>
  </div>
</DocumentFragment>
`;
